linera-service/src/linera-exporter/state.rs

```diff
@@ -1,21 +1,87 @@
 // Copyright (c) Zefchain Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-use linera_base::{crypto::CryptoHash, data_types::BlockHeight, identifiers::ChainId};
+use std::collections::{BTreeSet, VecDeque};
+
+use async_trait::async_trait;
+use linera_base::{
+    crypto::CryptoHash,
+    data_types::{Blob as ConfirmedBlob, BlockHeight},
+    identifiers::{BlobId, ChainId},
+};
+use linera_chain::types::ConfirmedBlock;
 use linera_client::config::DestinationId;
 use linera_sdk::views::{RootView, View, ViewError};
+use linera_storage::Storage;
 use linera_views::{
     context::Context, map_view::MapView, reentrant_collection_view::ReentrantCollectionView,
     register_view::RegisterView, views::ClonableView,
 };
+use serde::{Deserialize, Serialize};
 
+#[cfg(with_testing)]
+use crate::exporter_service::Summary;
 use crate::ExporterError;
 
 /// State of the linera exporter as a view.
 #[derive(Debug, RootView, ClonableView)]
-pub struct BlockExporterStateView<C> {
+pub(super) struct BlockExporterStateView<C> {
     /// The chain status, by chain ID.
     state: ReentrantCollectionView<C, ChainId, ChainStatusView<C>>,
+    /// The block(and blob) index.
+    index: MapView<C, Key, Value>,
+}
+
+#[derive(Debug, Clone, Serialize, Ord, PartialOrd, Eq, PartialEq, Deserialize, Hash)]
+pub(super) enum Key {
+    Block(CryptoHash),
+    Blob(BlobId),
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub(super) struct Block {
```

Comment:
`Block`, `Blob`, Batch` are used in different parts of the codebase with a different meaning. Can we use something different, maybe `ExporterBlock`, `ExporterBlob`, ExporterBatch` if we don't have any better idea?